client/reader_slice_of_slices_iterator.go

```diff
@@ -111,11 +114,34 @@ func (it *readerSliceOfSlicesIterator) CurrentLen() int {
 	return len(it.segments[it.idx].Unmerged)
 }
 
-func (it *readerSliceOfSlicesIterator) CurrentAt(idx int) io.Reader {
+func timeConvert(ticks *int64) time.Time {
+	if ticks == nil {
+		return time.Time{}
+	}
+	return xtime.FromNormalizedTime(*ticks, time.Second)
```

Comment:
Hm, we probably should be using nanoseconds here, in case block start and end times are at millisecond resolution for instance.

Reply:
Sure, sounds good. I thought we were truncating to seconds, but I guess it's better to lose precision as late as possible rather than early on :)